### DISCUSSION TITLE
Getting spring boot actuator metrics via OTEL library

### DISCUSSION THREAD

**Discussion by hshah2811** [signalfx/splunk-otel-java#1058] [Q&A]
I am using latest Splunk Otel Java library version Release v1.19.0 in my spring boot application. 

below is the configuration for java agent.

```
            - name: "OTEL_SERVICE_NAME"
              value: "service-name"
            - name: "OTEL_TRACES_EXPORTER"
              value: "jaeger-thrift-splunk"
            - name: "OTEL_EXPORTER_JAEGER_ENDPOINT"
              value: "http://<agent_host>:9080/v1/trace"
            - name: "SPLUNK_METRICS_ENDPOINT"
              value: "http://<agent_host>:9080/v2/datapoint"
            - name: "SPLUNK_METRICS_ENABLED"
              value: "true"
            - name: "OTEL_JAVAAGENT_ENABLED"
              value: "true"
            - name: OTEL_JAVAAGENT_DEBUG
              value: "true"
```
We are using SignalFx platform for monitoring. I can see the traces, spans of my application on signalFx. but I can see only limited metrics mentioned on this page 
[https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/java-otel-metrics-attributes.html ](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/java-otel-metrics-attributes.html )

however my spring boot actuator endpoint exposes many other metrics.  "/actuator/metrics"
`{
   "names":[
      "application.ready.time",
      "application.started.time",
      "disk.free",
      "disk.total",
      "executor.active",
      "executor.completed",
      "executor.pool.core",
      "executor.pool.max",
      "executor.pool.size",
      "executor.queue.remaining",
      "executor.queued",
      "http.server.requests",
      "jvm.buffer.count",
      "jvm.buffer.memory.used",
      "jvm.buffer.total.capacity",
      "jvm.classes.loaded",
      "jvm.classes.unloaded",
      "jvm.gc.live.data.size",
      "jvm.gc.max.data.size",
      "jvm.gc.memory.allocated",
      "jvm.gc.memory.promoted",
      "jvm.gc.overhead",
      "jvm.gc.pause",
      "jvm.memory.committed",
      "jvm.memory.max",
      "jvm.memory.usage.after.gc",
      "jvm.memory.used",
      "jvm.threads.daemon",
      "jvm.threads.live",
      "jvm.threads.peak",
      "jvm.threads.states",
      "logback.events",
      "process.cpu.usage",
      "process.files.max",
      "process.files.open",
      "process.start.time",
      "process.uptime",
      "rabbitmq.acknowledged",
      "rabbitmq.acknowledged_published",
      "rabbitmq.channels",
      "rabbitmq.connections",
      "rabbitmq.consumed",
      "rabbitmq.failed_to_publish",
      "rabbitmq.not_acknowledged_published",
      "rabbitmq.published",
      "rabbitmq.rejected",
      "rabbitmq.unrouted_published",
      "system.cpu.count",
      "system.cpu.usage",
      "system.load.average.1m",
      "tomcat.sessions.active.current",
      "tomcat.sessions.active.max",
      "tomcat.sessions.alive.max",
      "tomcat.sessions.created",
      "tomcat.sessions.expired",
      "tomcat.sessions.rejected"
   ]
}`

Example, I can not find "http.server.requests" of my app on signalFX. What configuration should I have to do in my app to push spring boot actuator metrics? 

let me know if need more info.

Note: Our infra has yet to configure OTEL collector hence we have not configured OTEL against OTEL_TRACES_EXPORTER.

**Comment by mateuszrzeszutek** [comment]
Hey @hshah2811 ,

Unfortunately the javaagent right now only supports the global registry (`Metrics.globalRegistry`). AFAIR Spring Boot does not use the global registry to create new instruments.

You could work around it by adding a bean that exposes the registry added by the javaagent to the application, e.g.:
```java
@Bean
MeterRegistry splunkMeterRegistry() {
  return Metrics.globalRegistry.getRegistries().iterator().next();
}
```
